File: LibraryProject/media.cpp
#include<iostream>
#include<string>
#include<vector>
#include"media.h"
#include "boost/algorithm/string.hpp"

using namespace std;
using namespace boost;

Media::Media(const string& a,const string& b,const string& c,const string& d)
{
    call_number = a;
    title = b;
    subjects = c;
    notes = d;
}

Media::~Media()
{

}

void Media::display() const
{

}

bool Media::compare_title(const string& searchby) const
{
    size_t found = title.find(searchby);
    if(found!=string::npos)
    {
        return true;
    }
    else
    {
        return false;
    }
}

bool Media::compare_call_number(const string& searchby) const
{
    size_t found = call_number.find(searchby);
    if(found!=string::npos)
    {
        return true;
    }
    else
    {
        return false;
    }
}

bool Media::compare_subject(const string& searchby) const
{
    size_t found = subjects.find(searchby);
    if(found!=string::npos)
    {
        return true;
    }
    else
    {
        return false;
    }
}

bool Media::compare_other(const string& x) const
{

}

File: Python-Library-Project/media.py
#Sumeet Tulsani
class Media:
    def __init__(self,a,b,c,d):
        self.call_number = a;
        self.title = b;
        self.subjects = c;
        self.notes = d;

    def display():
        pass

    def compare_title(self,searchby):
        if self.title.find(searchby) != -1:
            return True
        else:
            return False

    def compare_call_number(self,searchby):
        if self.call_number.find(searchby) != -1:
            return True
        else:

            return False

    def compare_subject(self,searchby):
        if self.subjects.find(searchby) != -1:
            return True
        else:
            return False

    def compare_other(searchby):
        pass
